src/lock_file/package_identifier.rs

```diff
@@ -118,23 +117,20 @@ impl PypiPackageIdentifier {
         })
     }
 
-    pub fn satisfies(&self, requirement: &Requirement) -> bool {
+    pub fn satisfies(&self, requirement: &pypi_types::Requirement) -> bool {
         // Verify the name of the package
         if self.name.as_normalized() != &requirement.name {
             return false;
         }
 
         // Check the version of the requirement
-        match &requirement.version_or_url {
-            None => true,
-            Some(VersionOrUrl::Url(url)) => {
-                // Check if the URL matches
-                url.to_url() == self.url
-            }
-            Some(VersionOrUrl::VersionSpecifier(required_spec)) => {
-                // Check if the locked version is contained in the required version specifier
-                required_spec.contains(&self.version)
+        match &requirement.source {
+            pypi_types::RequirementSource::Registry { specifier, .. } => {
+                specifier.contains(&self.version)
             }
+            pypi_types::RequirementSource::Url { url, .. } => url.raw() == &self.url,
+            pypi_types::RequirementSource::Git { .. } => false,
+            pypi_types::RequirementSource::Path { .. } => false,
```

Comment:
Shouldn't we check this as well? Or do we always store those as Urls? Or are these always out of date?

Reply:
Excellent question! I think the url check should be also false here because we verified it here: 
https://github.com/prefix-dev/pixi/blob/main/src/lock_file/satisfiability.rs#L606

Reply:
Should we maybe add unreachable! then to be certain?

Reply:
Changed it!